### ISSUE TITLE
 Discrepancies in CSL-Daily Dataset Results

### ISSUE THREAD

**Issue by 486715538** [sutwangyan/MSKA#4]
Dear Sir/Madam,

I am in the process of replicating your code, and I have encountered some discrepancies in the results for the CSL-Daily dataset. The word error rate (WER) I obtained is 28.2382, which is notably higher than the anticipated 27.1.

Would it be possible for you to kindly share if there are any variations in parameter settings that I might have overlooked or need to adjust?

The command I am utilizing for the code is as follows:

python train.py --config configs/${dataset}_s2g.yaml --epoch 100
After conducting 100 epochs of training, the test outcomes are detailed below:

Averaged statistics: loss: 47.4283 (62.4941), WER: 28.2382 (28.2382)
DEV loss: 62.494
Test loss of the network on the 147 test videos: 62.494
Training time: 20 hours, 19 minutes, and 26 seconds
I would greatly appreciate any guidance or insights you could provide regarding these results.

Thank you very much for your time and assistanc

**Comment by sutwangyan** [comment]
Hello, during our training, we perform multiple runs to select the best result. In fact, even when the same random seed is used, randomness in the results may still occur. The reason might be that certain torch operations, such as torch.nn.CTCLoss, involve non-deterministic behaviors. Please refer to [https://pytorch.org/docs/stable/generated/torch.use_deterministic_algorithms.html#torch.use_deterministic_algorithms]

**Comment by 486715538** [comment]
I'm delighted to receive your reply.
I have a question regarding the achievement of the 27.1 result. Were the hyperparameters you set the same as those provided on GitHub? For instance, the learning rate, number of epochs, and random seed.

**Comment by sutwangyan** [comment]
Yes, the parameters are the same
